### PR TITLE
Server TypeScript refactor

### PR DESCRIPTION
**Pull request by sbuggay**
Converting both server-data and server-host to use `ts-node` so they can be written in TypeScript and share interfaces.

### CODE REVIEW

File: packages/sentry-client-web/src/components/List.tsx [hunk 1]
```diff
@@ -80,20 +91,29 @@ export class List extends React.Component<IListProps, IListState> {
...
-                            <Service
-                                key={index}
-                                service={service} />
-                        </div>
-                    );
-                })}
-            </div>
-        );
-
+        if (services.length > 0) {
```
Comment (jfo84) [inline packages/sentry-client-web/src/components/List.tsx]: The code here for rendering servers and services is nearly identical. You could have a component instead that has an interface of an array of `displayEntities`, servers or services. Generalizing the null handling code for servers is probably fine, if not better

File: packages/sentry-client-web/src/components/MessageBox.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    textStyle?: React.CSSProperties;
+}
+
+export class MessageBox extends React.Component<IMessageBoxProps, any> {
+    public getStyle(): React.CSSProperties {
+        return {
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "center",
+            ...this.props.style
```
Comment (jfo84) [inline packages/sentry-client-web/src/components/MessageBox.tsx]: The flexible `style` interface is cool but definitely not needed. It clutters up `List`

File: packages/sentry-server-data/src/Cache.ts [hunk 3]
```diff
@@ -1,34 +1,43 @@
...
 }
 
-class Cache {
+interface IStore {
+    [key: string]: any
+}
+
+export default class Cache {
+    store: IStore;
+    cacheFunctions: any;
```
Comment (jfo84) [inline packages/sentry-server-data/src/Cache.ts]: `Object`?

File: packages/sentry-server-data/src/Cache.ts [hunk 4]
```diff
@@ -1,34 +1,43 @@
...
     /** Clear the cache. */
     clear() {
         this.store = {};
-        this.intervalFunctions = {};
+        this.cacheFunctions = {};
     }
 
     /** Set the config by assigning it to the default config. */
-    setConfig(config) {
+    setConfig(config: any) {
```
Comment (jfo84) [inline packages/sentry-server-data/src/Cache.ts]: `Object`?

File: packages/sentry-server-data/src/Cache.ts [hunk 5]
```diff
@@ -38,42 +47,37 @@ class Cache {
...
-    addIntervalFunction(key, callback) {
-        this.intervalFunctions[key] = callback;
+    addCacheFunction(key: string, callback: Function) {
+        this.cacheFunctions[key] = callback;
     }
 
      /** Add a function array to the function queue. */
-    addIntervalFunctionArray(key, array) {
-        this.intervalFunctions[key] = array;
+    addCacheFunctions(key: string, array: Function[]) {
```
Comment (jfo84) [inline packages/sentry-server-data/src/Cache.ts]: Can you do `Function[]`? I can't find anything like that by googling

File: packages/sentry-server-data/src/Server.ts [hunk 6]
```diff
@@ -27,9 +28,9 @@ const dynamicInfo = () => {
     });
 }
 
-const serviceInfo = () => {
+export function serviceInfo() {
```
Comment (jfo84) [inline packages/sentry-server-data/src/Server.ts]: why not `export const serviceInfo = () => {`

File: packages/sentry-server-data/src/actionHandler.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+            break;
+        default:
+            break;
+    }
+}
+
+
+
+export function addService() {
```
Comment (jfo84) [inline packages/sentry-server-data/src/actionHandler.ts]: Nice!

File: packages/sentry-server-data/src/actionHandler.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+const SERVICE_REMOVE = "remove-service";
+const SERVICE_EDIT = "edit-service";
+const SERVICE_LIST = "list-services";
+
+import { updateServices } from "./configHandler";
+
+
+prompt.message = "";
+
+export function handleAction(args: string[]) {
```
Comment (jfo84) [inline packages/sentry-server-data/src/actionHandler.ts]: ES6?

File: packages/sentry-server-data/src/configHandler.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+const json = require("json-update");
+
+const configFile = "config.json";
+
+export async function update(value: Object) {
```
Comment (jfo84) [inline packages/sentry-server-data/src/configHandler.ts]: ES6 for these two?

File: packages/sentry-server-host/src/api.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import * as express from "express";
+let router = express.Router();
+
+import { pollServers } from "./polling";
+
+router.get("/", (req, res) => {
+	res.header("Access-Control-Allow-Origin", "*");
+	res.header("Content-Type: application/json");
+	let hosts = [
+		"http://127.0.0.1:3333",
```
Comment (jfo84) [inline packages/sentry-server-host/src/api.ts]: Why the duplication here?

File: packages/sentry-server-host/src/configHandler.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import * as nconf from "nconf";
+
+export function configInit(path: string) {
```
Comment (jfo84) [inline packages/sentry-server-host/src/configHandler.ts]: ES6?

File: packages/sentry-server-data/src/Server.ts [hunk 12]
```diff
@@ -27,9 +28,9 @@ const dynamicInfo = () => {
...
     const services = config.services;
-    let serviceFunctions = [];
+    let serviceFunctions: any[] = [];
```
Comment (jfo84) [inline packages/sentry-server-data/src/Server.ts]: If `Function[]` works then here too

File: packages/sentry-server-data/src/actionHandler.ts (hunk 8 above)
Comment (sbuggay) [inline packages/sentry-server-data/src/actionHandler.ts]: No typings have been made for `prompt` yet

**Pull request merged**